### PR TITLE
Updating GLM pipeline with new lakes 

### PR DESCRIPTION
**Pull request by jzwart**
Resurrecting this pipeline for our lake temperature sprint. 

- Updated lake list and shapefiles from [pipeline #1](https://github.com/USGS-R/lake-temperature-model-prep)
- Set up makefile for generating `nml` files for running GLM. I don't think it makes sense to create an `ind` file for these so am just creating these in a tmp dir, which will get rsync-ed to Yeti. 
- Copied over some Yeti functions that Alison created. These should help with automatically syncing the pipeline to Yeti. Updated README with some instructions on how to set up ssh with Yeti.

### CODE REVIEW

**Comment by jzwart** [comment]
Based on building the ~100 `nml` files, it should take about 95 mins to generate all ~19,000 `nml` files on one computer locally

**Comment by limnoliver** [comment]
@jzwart in the past we've ignored task remake files. It looks like Alison also does this in pipeline three using a similar combiner model. I think you'll want to add `1_nml_tasks_makefile.yml` to `.gitignore`?

File: lib/cfg/settings.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# file paths on Yeti
+drivers_yeti_path: '/cxfs/projects/usgs/water/iidd/data-sci/lake-temp/driver-data'
+drivers_time: '0.350500'
```
Comment (jordansread) [inline lib/cfg/settings.yml:7]: I think this is fine, but note that if we change the driver timestep, we'll need to remember to change this one too.

File: lib/src/utils.R [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+#'   src_dir)
+yeti_get <- function(src_dir, dest_dir, files) {
+  user <- Sys.info()[['user']]
+  if(length(src_dir) != 1) stop('Need exactly one unique src_dir per rsync call (as implemented)')
+  src_path <- sprintf('%s@yeti.cr.usgs.gov:%s', user, src_dir)
+  if(!dir.exists(dest_dir)) dir.create(dest_dir, recursive=TRUE)
+  tmpfile <- file.path(dest_dir, '_temp_rsync_file_list.txt') # must be a relative path to work on Windows
+  readr::write_lines(files, tmpfile)
+  on.exit(file.remove(tmpfile))
+  syncr::rsync(src=src_path, dest=dest_dir, files_from=tmpfile)
```
Comment (jordansread) [inline lib/src/utils.R]: oh wow, this works? handy

**Review by jordansread — COMMENTED** [review]
Looks good to me, but wondering which scipiper version are you using? Is it the one w/ the task_combiners? or previous version?

**Comment by jordansread** [comment]
Agreed w/ Sam on the task makefiles, especially if they are this big. (I think it is fine to include small ones if it makes sense to).

File: lib/cfg/settings.yml (hunk 1 above)
Comment (jzwart) [reply lib/cfg/settings.yml:7]: Yup sounds good. I'll make a note if I use this cfg - right now it's just a placeholder.

File: lib/src/utils.R (hunk 2 above)
Comment (jzwart) [reply lib/src/utils.R]: Haven't tried it yet, just copied over from Alison's workflow for PGDL for now. Will try out tomorrow

**Comment by jzwart** [comment]
> Looks good to me, but wondering which scipiper version are you using? Is it the one w/ the task_combiners? or previous version?

I'm a little unclear if I'm using the combiner version of `scipiper`. I updated this morning but with the `master` branch. Looks like there is `task_combiner` branch of scipiper, which I am _not_ using

**Comment by jordansread** [comment]
@limnoliver what are you using? 

I am using the combiner branch for my work on the temperature prep pipeline...I think we should all use the same version, but I understand that the jump to the combiners version is not trivial (hence it is still hanging out on a branch). 

If you are willing to make the jump, it will help us simplify some of the solutions and also help us from colliding w/ different versions on different parts of the pipeline.

**Comment by limnoliver** [comment]
I was using the latest version of the master branch. But yes, I think we should switch to the combiners branch since @jzwart is using that workflow as well.

**Comment by jzwart** [comment]
Sounds good, I'll switch to the combiners branch

**Comment by jordansread** [comment]
Sorry, I didn't see that this was still open until now. I am working on a similar pattern, but now that we have nhd HR, some of the functions won't work (e.g., `populate_base_nml()`)

**Pull request merged**